### PR TITLE
Move to Rails 6.0 defaults

### PR DESCRIPTION
**Pull request by duncanjbrown**
### Context

We're still on Rails 5.2 defaults and we're starting to see deprecation warnings.

Move up to 6.0 as a first step.

### Changes proposed in this pull request

Take on all the 6.0 defaults but one.

### Guidance to review

See comments inline on each new setting.

Use the review app to make sure everything still works.

### CODE REVIEW

File: config/initializers/new_framework_defaults_5_2.rb [hunk 1]
```diff
@@ -1,39 +0,0 @@
...
-#
-# This option is not backwards compatible with earlier Rails versions.
-# It's best enabled when your entire app is migrated and stable on 5.2.
-#
-# Existing cookies will be converted on read then written with the new scheme.
-# Rails.application.config.action_dispatch.use_authenticated_cookie_encryption = true
-
-# Use AES-256-GCM authenticated encryption as default cipher for encrypting messages
-# instead of AES-256-CBC, when use_authenticated_message_encryption is set to true.
-# Rails.application.config.active_support.use_authenticated_message_encryption = true
```
Comment (duncanjbrown) [inline config/initializers/new_framework_defaults_5_2.rb:25]: We don't use this feature

File: config/initializers/new_framework_defaults_5_2.rb [hunk 2]
```diff
@@ -1,39 +0,0 @@
...
-
-# Add default protection from forgery to ActionController::Base instead of in
-# ApplicationController.
-# Rails.application.config.action_controller.default_protect_from_forgery = true
```
Comment (duncanjbrown) [inline config/initializers/new_framework_defaults_5_2.rb:29]: This doesn't change anything for us

File: config/initializers/new_framework_defaults_5_2.rb [hunk 3]
```diff
@@ -1,39 +0,0 @@
...
-
-# Store boolean values are in sqlite3 databases as 1 and 0 instead of 't' and
-# 'f' after migrating old data.
-# Rails.application.config.active_record.sqlite3.represent_boolean_as_integer = true
```
Comment (duncanjbrown) [inline config/initializers/new_framework_defaults_5_2.rb:33]: No AR

File: config/initializers/new_framework_defaults_5_2.rb [hunk 4]
```diff
@@ -1,39 +0,0 @@
...
-
-# Use SHA-1 instead of MD5 to generate non-sensitive digests, such as the ETag header.
-# Rails.application.config.active_support.use_sha1_digests = true
```
Comment (duncanjbrown) [inline config/initializers/new_framework_defaults_5_2.rb:36]: Might break some caches, but we hardly client side cache anyway

File: config/initializers/new_framework_defaults_6_0.rb [hunk 5]
```diff
@@ -1,46 +0,0 @@
...
-# Be sure to restart your server when you modify this file.
-#
-# This file contains migration options to ease your Rails 6.0 upgrade.
-#
-# Once upgraded flip defaults one by one to migrate to the new default.
-#
-# Read the Guide for Upgrading Ruby on Rails for more info on each option.
-
-# Don't force requests from old versions of IE to be UTF-8 encoded.
-# Rails.application.config.action_view.default_enforce_utf8 = false
```
Comment (duncanjbrown) [inline config/initializers/new_framework_defaults_6_0.rb:11]: Not harmful

File: config/initializers/new_framework_defaults_6_0.rb [hunk 6]
```diff
@@ -1,46 +0,0 @@
...
-
-# Embed purpose and expiry metadata inside signed and encrypted
-# cookies for increased security.
-#
-# This option is not backwards compatible with earlier Rails versions.
-# It's best enabled when your entire app is migrated and stable on 6.0.
-# Rails.application.config.action_dispatch.use_cookies_with_metadata = true
-
-# Change the return value of `ActionDispatch::Response#content_type` to Content-Type header without modification.
-# Rails.application.config.action_dispatch.return_only_media_type_on_content_type = false
```
Comment (duncanjbrown) [inline config/initializers/new_framework_defaults_6_0.rb:21]: Not harmful

File: config/initializers/new_framework_defaults_6_0.rb [hunk 7]
```diff
@@ -1,46 +0,0 @@
...
-
-# Return false instead of self when enqueuing is aborted from a callback.
-# Rails.application.config.active_job.return_false_on_aborted_enqueue = true
```
Comment (duncanjbrown) [inline config/initializers/new_framework_defaults_6_0.rb:24]: Not harmful

File: config/initializers/new_framework_defaults_6_0.rb [hunk 8]
```diff
@@ -1,46 +0,0 @@
...
-
-# Send Active Storage analysis and purge jobs to dedicated queues.
-# Rails.application.config.active_storage.queues.analysis = :active_storage_analysis
-# Rails.application.config.active_storage.queues.purge    = :active_storage_purge
```
Comment (duncanjbrown) [inline config/initializers/new_framework_defaults_6_0.rb:28]: Not harmful

File: config/initializers/new_framework_defaults_6_0.rb [hunk 9]
```diff
@@ -1,46 +0,0 @@
...
-
-# When assigning to a collection of attachments declared via `has_many_attached`, replace existing
-# attachments instead of appending. Use #attach to add new attachments without replacing existing ones.
-# Rails.application.config.active_storage.replace_on_assign_to_many = true
```
Comment (duncanjbrown) [inline config/initializers/new_framework_defaults_6_0.rb:32]: We don't use AS

File: config/initializers/new_framework_defaults_6_0.rb [hunk 10]
```diff
@@ -1,46 +0,0 @@
...
-
-# Use ActionMailer::MailDeliveryJob for sending parameterized and normal mail.
-#
-# The default delivery jobs (ActionMailer::Parameterized::DeliveryJob, ActionMailer::DeliveryJob),
-# will be removed in Rails 6.1. This setting is not backwards compatible with earlier Rails versions.
-# If you send mail in the background, job workers need to have a copy of
-# MailDeliveryJob to ensure all delivery jobs are processed properly.
-# Make sure your entire app is migrated and stable on 6.0 before using this setting.
-# Rails.application.config.action_mailer.delivery_job = "ActionMailer::MailDeliveryJob"
```
Comment (duncanjbrown) [inline config/initializers/new_framework_defaults_6_0.rb:41]: We don't use AM!

File: config/initializers/new_framework_defaults_6_0.rb [hunk 11]
```diff
@@ -1,46 +0,0 @@
...
-
-# Enable the same cache key to be reused when the object being cached of type
-# `ActiveRecord::Relation` changes by moving the volatile information (max updated at and count)
-# of the relation's cache key into the cache version to support recycling cache key.
-# Rails.application.config.active_record.collection_cache_versioning = true
```
Comment (duncanjbrown) [inline config/initializers/new_framework_defaults_6_0.rb:46]: Not harmful

File: config/initializers/new_framework_defaults_6_0.rb [hunk 12]
```diff
@@ -1,46 +0,0 @@
...
-
-# Don't force requests from old versions of IE to be UTF-8 encoded.
-# Rails.application.config.action_view.default_enforce_utf8 = false
-
-# Embed purpose and expiry metadata inside signed and encrypted
-# cookies for increased security.
-#
-# This option is not backwards compatible with earlier Rails versions.
-# It's best enabled when your entire app is migrated and stable on 6.0.
-# Rails.application.config.action_dispatch.use_cookies_with_metadata = true
```
Comment (duncanjbrown) [inline config/initializers/new_framework_defaults_6_0.rb:18]: Kept the old behaviour in `application.rb`

**Review by tomas-stefano — APPROVED** [review]

**Pull request merged**